eval/vllm_runner.py

```diff
@@ -97,35 +98,46 @@ def run_vllm_eval(args):
             axis=1,
         )
         print(f"Prepared {len(df)} question(s) from {questions_file}")
-        print(f"Generating completions")
-        start_time = time.time()
-        # we pass the full list of prompts at once to the vllm engine
-        outputs = llm.generate(df["prompt"].tolist(), sampling_params)
-        time_taken = time.time() - start_time
-        print(f"Time taken: {time_taken:.1f}s")
 
-        # save generation metrics
-        df["latency_seconds"] = time_taken / len(df)
+        def chunk_dataframe(df, chunk_size):
+            """Returns successive chunk_size chunks from df as a list of dfs"""
+            df_chunks = []
+            for i in range(0, len(df), chunk_size):
+                df_i = df.iloc[i : min(i + chunk_size, len(df))]
+                print(
+                    f"Chunk {i//chunk_size+1}/{len(df)//chunk_size+1} with {len(df_i)} questions"
+                )
+                df_chunks.append(df_i)
+            return df_chunks
+
+        df_chunks = chunk_dataframe(df, args.batch_size)
 
-        df["generated_query"] = ""
-        df["tokens_used"] = 0
-        df["correct"] = 0
-        df["exact_match"] = 0
-        df["error_db_exec"] = 0
-        df["error_msg"] = ""
+        total_tried = 0
         total_correct = 0
-        with tqdm(total=len(df)) as pbar:
-            for i, output in enumerate(outputs):
+        output_rows = []
+
+        print(f"Generating completions")
+
+        for batch in (pbar := tqdm(df_chunks, total=len(df))):
+            prompts = batch["prompt"].tolist()
+            print(f"Generating completions for {len(prompts)} prompts")
+            start_time = time.time()
+            outputs = llm.generate(prompts, sampling_params)
+            print(
+                f"Generated {len(outputs)} completions in {time.time() - start_time:.2f} seconds"
+            )
+            time_taken = time.time() - start_time
+            for row, output in zip(batch.to_dict("records"), outputs):
                 generated_query = (
                     output.outputs[0].text.split(";")[0].split("```")[0].strip() + ";"
                 )
                 normalized_query = sqlparse.format(
                     generated_query, keyword_case="upper", strip_whitespace=True
                 )
-                df.loc[i, "generated_query"] = normalized_query
-                df.loc[i, "tokens_used"] = len(output.outputs[0].token_ids)
-                df.loc[i, "latency_seconds"] = time_taken / len(df)
-                row = df.iloc[i]
+                row["generated_query"] = normalized_query
+                row["tokens_used"] = len(output.outputs[0].token_ids)
+                row["latency_seconds"] = time_taken / len(batch)
```

Comment:
With batching, is it a downside that the result of latency_seconds won't be as useful?

Reply:
Yup! This is also why we've removed it from the `hf_runner`. It's much more useful to track if we're using a batch size of 1, though

Reply:
yep - it has been batching everything as a single batch all this while so the latency_seconds for vllm hasn't been particularly useful.